### PR TITLE
Adding nodes linked to menu commands

### PR DESCRIPTION
**Pull request by EdoFro**
Hi Dimitry,

This PR replaces #1536

>Your current solution will not work on MacOS because on MacOS the menu is implemented by the OS and SwingUtilities will not find it. I would actually add "Insert node linked to menu command" as the new action and implement is similar to SetAcceleratorOnNextClickAction and implementing extending the logic in AccelerateableAction.actionPerformed(ActionEvent).
>
>It would make the menu look better (I like a menu item name "Insert node linked to menu command" more than "Set 'Insert node linked to menu command' hot key") and allow it to work on Macs too.

I changed the whole logic and now it works similar to SetAcceleratorOnNextClickAction 👍🏼 

I added the new feature in `menu-> insert->node` and in `menu-> insert->link` 

It works also when clicking a menu command and holding Shift + Ctrl keys.

I also added that when calling the feature through the menu command, it opens a dialog. This dialog stays open until the user closes it. While this dialog is open the user can insert multiple nodes and the menus stay open to easy navigation.

I hope this time is better.

Bye,

edo

### CODE REVIEW

**Comment by dpolivaev** [comment]
Could you please check my changes? I tried to simplify the logic even further and to make it more consistent. Also, I removed control+shift as a hot key because you can assign any hot key to the new action and use it to activate the action.

**Comment by EdoFro** [comment]
Ok. I think you simplified a little too much.

When inserting new "nodes linked to menu items", I mostly want to add a group of them, to make me a custom "Menu" of features in a branch.

It's hard for me to explain it, so I made two videos to show the difference between both versions.

You don't need to see them until the end. It's just to get the idea.


## Your version :

https://github.com/freeplane/freeplane/assets/35700575/d2235cba-20b2-4819-8486-338aed55ed2f


## My version:

https://github.com/freeplane/freeplane/assets/35700575/eaad6d9f-5543-49cb-a84c-170d399cd718


Tell me what you think.

edo

**Comment by EdoFro** [comment]
> Also, I removed control+shift as a hot key because you can assign any hot key to the new action and use it to activate the action.

I think you're right. It was not really practical.

**Comment by dpolivaev** [comment]
I got it. Please check my changes, I explained my reasons in the commit messages.

**Review by EdoFro — COMMENTED** [review]
Edit:
I was reviewing each commit and tried to add a comment to one of them, but the comment ended here in the conversation.
I thought each commit could had its own conversation. sorry.

**Comment by EdoFro** [comment]
> I got it. Please check my changes, I explained my reasons in the commit messages.

I took a look at the changes and here are my comments:



|Summary|Commit| my comments |
|-----|-----|-----|
revert unrelated change | [bee3ad6](https://github.com/freeplane/freeplane/pull/1543/commits/bee3ad687595c4cb45acc7112391b229f23c50d4) | This IS related! <br>I don't understand why you revert this.
remove comment | [7941ad0](https://github.com/freeplane/freeplane/pull/1543/commits/7941ad0dc73dd7ddf3b62e3ebb8b5c3fc1ce5c1d) |👍🏼 
make button text on cancel dialog configurable | [d93dda1](https://github.com/freeplane/freeplane/pull/1543/commits/d93dda1a5fc09ce084b00259160b57892722eeb0) |👍🏼 
Define text string "done" copied from simplyhtml.done | [727e22d](https://github.com/freeplane/freeplane/pull/1543/commits/727e22dfe6bc852362a00d12d216ef695dbd655d) |👍🏼 
show "done" instead of "cancel", improve action name and dialog message | [efc648f](https://github.com/freeplane/freeplane/pull/1543/commits/efc648f3b749153963fb5b81643b2db391b6e756) | 👍🏼 
show new nodes | [15e235b](https://github.com/freeplane/freeplane/pull/1543/commits/15e235bc97eae76c52b749e81575e4da69f8f445) | 👍🏼 
FP fix navigation to siblings: root can't be a sibling | [ddd3a02](https://github.com/freeplane/freeplane/pull/1543/commits/ddd3a02fe221377575d9a661263c274df5bb47a3) | 🤷🏼 No idea

File: freeplane/src/main/java/org/freeplane/core/ui/components/JFreeplaneMenuItem.java
Comment (EdoFro) [inline freeplane/src/main/java/org/freeplane/core/ui/components/JFreeplaneMenuItem.java:1]: The additional processMouseEvent method allows to click the menu items one after another without the menu to close with every click.

Comment (EdoFro) [reply freeplane/src/main/java/org/freeplane/core/ui/components/JFreeplaneMenuItem.java:1]: in this video you can see that the icon menu stays visible after clicking each of the menu Items (Zebra, Penguin, Racoon, ..):

https://github.com/freeplane/freeplane/assets/35700575/eaad6d9f-5543-49cb-a84c-170d399cd718

Comment (dpolivaev) [reply freeplane/src/main/java/org/freeplane/core/ui/components/JFreeplaneMenuItem.java:1]: Thanks, I got it and merged your version. See https://github.com/freeplane/freeplane/pull/1543/commits/913657ff254874589e50ec9821258394351947c7. The unrelated commit was by mistake, it belongs to another feature and not on your branch.

**Pull request merged**